### ISSUE TITLE
Memory leak when building cpp using optimization (O3)

### ISSUE THREAD

**Issue by patrickhart** [bark-simulator/bark#107]
When the -O3 flag is set in the .bazelrc, a memory leak occures in several tests.

.bazelrc
```
test --test_output=errors --action_env="GTEST_COLOR=1"

# Force bazel output to use colors (good for jenkins) and print useful errors.
common --color=yes

build --cxxopt='-std=c++14'
```

**Comment by xmyqsh** [comment]
A memory leak solver PR is here #101

**Comment by patrickhart** [comment]
Fixed in #108.
